common/plugins/eu.esdihumboldt.hale.common.schema.persist/src/eu/esdihumboldt/hale/common/schema/persist/hsd/MergeSchemas.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 wetransform GmbH
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.hale.common.schema.persist.hsd;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Streams;
+
+import eu.esdihumboldt.hale.common.schema.model.Schema;
+import eu.esdihumboldt.hale.common.schema.model.impl.DefaultSchemaSpace;
+
+/**
+ * Helper class for merging multiple schemas for export.
+ * 
+ * @author Simon Templer
+ */
+public class MergeSchemas {
+
+	private static class MergedSchema extends DefaultSchemaSpace implements Schema {
+
+		private final String namespace;
+
+		public MergedSchema(String namespace, Iterable<? extends Schema> schemas) {
+			this.namespace = namespace;
+			for (Schema schema : schemas) {
+				addSchema(schema);
+			}
+		}
+
+		@Override
+		public URI getLocation() {
+			throw new UnsupportedOperationException();
+		}
+
+		@Override
+		public String getNamespace() {
+			return this.namespace;
+		}
+
+	}
+
+	/**
+	 * Namespace to use for combined schemas if no common namespace is
+	 * available.
+	 */
+	public static final String DEFAULT_COMBINED_NAMESPACE = HaleSchemaConstants.NS + "/combine";
```

Comment:
What happens if the combined schema is then again merged with some other schema? For instance importing it again in HS with some other schema
- Is this scenario possible? 
- Will this lead to `"http://www.esdi-humboldt.eu/hale/schema/combined/combined" namespace?

Reply:
Good point. For the HSD export it would not matter though, because the namespaces of the schemas that are combined are not retained other than their usage in the defined types. And that is where the combined namespace is not used. So I think we are safe with the current approach.